### PR TITLE
Added include_data parameter & changed only_monitored default to true

### PR DESCRIPTION
**Pull request by liiight**
Added ability to set show quality via Sonarr's 'cutoff' quality parameter.
Also changed the default of only_monitored to true

### CODE REVIEW

**Comment by liiight** [comment]
Could you point me in the direction of why this build failed? I couldn't make sense of it

File: flexget/plugins/input/sonarr_emit.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        'required': ['api_key', 'base_url'],
+        'additionalProperties': False
+    }
+
+    def get_page(self, task, config, page_number):  # Function that gets a page number and page size and returns the responding result json
+        parsedurl = urlparse(config.get('base_url'))
+        url = '%s://%s:%s%s/api/wanted/missing?page=%d&pageSize=%d&sortKey=series.title&sortdir=asc' % (parsedurl.scheme, parsedurl.netloc, config.get('port'),
+                                                                                                        parsedurl.path, page_number, config.get('page_size'))
+        headers = {'X-Api-Key': config['api_key']}
+        json = task.requests.get(url, headers=headers).json()
```
Comment (stevezau) [inline flexget/plugins/input/sonarr_emit.py]: Perhaps we need some exception handling here?

Comment (liiight) [inline flexget/plugins/input/sonarr_emit.py]: You're right obviously, i'll add it

File: flexget/plugins/input/sonarr_emit.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+                                  series_season=season,
+                                  series_episode=episode,
+                                  series_id='S%02dE%02d' % (season, episode),
+                                  tvdb_id=record['series']['tvdbId'],
+                                  tvrage_id=record['series']['tvRageId'],
+                                  title=record['series']['title'] + ' ' + 'S%02dE%02d' % (season, episode))
+                    if entry.isvalid():
+                        entries.append(entry)
+                    else:
+                        log.debug('Invalid entry created? %s' % entry)
```
Comment (stevezau) [inline flexget/plugins/input/sonarr_emit.py]: Maybe make this log.error?

File: flexget/plugins/input/sonarr_emit.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+                        log.info("    Title: %s" % entry["title"])
+                        log.info("    URL: %s" % entry["url"])
+                        log.info("    Show name: %s" % entry["series_name"])
+                        log.info("    TVDB ID: %s" % entry["tvdb_id"])
+                        log.info("    TVRAGE ID: %s" % entry["tvrage_id"])
+                        log.info("    Season: %s" % entry["series_season"])
+                        log.info("    Episode: %s" % entry["series_episode"])
+                        log.info("    Series ID: %s" % entry["series_id"])
+                    continue
+            json = self.get_page(task, config, page)
```
Comment (stevezau) [inline flexget/plugins/input/sonarr_emit.py]: Just a suggestion but perhaps it's better to build json then loop through it.. such as

json[pagenum][entries]

that way you can handle all exceptions for requests before you add the entries

Comment (liiight) [inline flexget/plugins/input/sonarr_emit.py]: You mean run get all data into local json and then operate on it in its entirety? I though of that but data can be potentially huge. Image dozens of shows, each one with several seasons of unwatched episodes.
Also note that there is a preliminary API call before the loop to get the total pages value. If there is a outstanding connection error that is derived from bad config, the plugin should stop there.

Comment (stevezau) [inline flexget/plugins/input/sonarr_emit.py]: Wouldn't it have to do it either way??

**Comment by stevezau** [comment]
@gazpachoking i'm not 100% sure on this one, code looks ok to me but would prefer you to review before i merge.

**Comment by liiight** [comment]
I appreciate the caution, it's always best to have more eyes look at everything, but as I said before, this PR contains a critical bug fix for the Sickbeard plugin, as it is currently not working (causes all download paths to be null).

**Comment by stevezau** [comment]
@liiight This PR seems to contain alot more then just a bug fix.. Perhaps you can open a new PR for only the bug fix? I'll merge that one in straight away.

**Comment by liiight** [comment]
Sound like a good idea, but i'm not sure how I do that. Can you help?

**Comment by stevezau** [comment]
@liiight might be faster if you point out the code fixes and ill do it manually

**Comment by liiight** [comment]
@stevezau you can just merge sickbeard.py then. It contains error handling and the bug fix (removal of any reference to 'path')

File: flexget/plugins/input/sonarr_emit.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+        except RequestException as e:
+            raise plugin.PluginError('Unable to connect to Sonarr at %s://%s:%s%s. Error: %s'
+                                     % (parsedurl.scheme, parsedurl.netloc, config.get('port'),
+                                        parsedurl.path, e))
+        return json
+
+    def on_task_input(self, task, config):
+        json = self.get_page(task, config, 1)
+        entries = []
+        pages = int(json['totalRecords'] / config.get('page_size')) + (json['totalRecords'] % config.get('page_size') > 0)  # Sets number of requested pages
```
Comment (gazpachoking) [inline flexget/plugins/input/sonarr_emit.py]: I think this would be nicer if we do `math.ceil(json['totalRecords'] / config.get('page_size'))`

Comment (liiight) [inline flexget/plugins/input/sonarr_emit.py]: I think there was a reason I did it like that, but I can't figure out why. I'll change this.

File: flexget/plugins/input/sonarr_emit.py (hunk 3 above)
Comment (liiight) [inline flexget/plugins/input/sonarr_emit.py]: But I wouldn't have to loop over everything twice (even though, 2nd loop is a local representation)

**Comment by liiight** [comment]
Any update with merging this PR?

**Comment by gazpachoking** [comment]
I would prefer the new plugin be in a separate PR to fixes next time, but it all looks good to me. Thanks!

**Comment by gazpachoking** [comment]
@liiight If you can update the wiki as needed that would be great. :smile:

**Comment by liiight** [comment]
Wiki already updated. I'll make sure new plugins will be separated next time, thanks!

**Pull request merged**